src/probe/probe_iterator.jl

```diff
@@ -45,7 +55,7 @@ function probe(examples::Vector{<:IOExample}, iterator::ProgramIterator, select:
             return nothing
         end
 
-        # partial_sols = select(partial_sols, eval_cache) # select promising partial solutions
+        partial_sols = selectpsol_largest_subset(psol_with_eval_cache) # select promising partial solutions
```

Comment:
Why is the selection function hardcoded? Shouldn't you use the `select` function that is passed to `probe`?
```suggestion
        partial_sols = select(psol_with_eval_cache) # select promising partial solutions
```

Reply:
Yes, you are right, the select function should be used.